.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -12,14 +12,11 @@ Before submitting your PR make sure that:
 * Your feature branch is up-to-date with `master` (if not - rebase it), and does not include merge commits.
 * Related commits are squashed together, and unrelated commits are splitted apart.
 * Your PR includes a regression test if it fixes a bug.
-* You add an entry to the [Changelog] if the new code introduces user-observable changes. See [changelog entry format].
```

Comment:
It would help to inform contributors that the PR title will be used in the changelog since its auto generated now. Perhaps a new line here to point that out or add on to existing/related line?

If I understand right, the PR needs to have an appropriate label now which is great. Perhaps a line to have them pick one of the `type: ...` labels they think is a best fit, at least to start/requirement to create PR?

Reply:
I was thinking of adding something about this, but:

* Regarding the PR title, we already have "The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.". I think that guideline might be good enough? Also, we can always change the title ourselves before merge, after merge, or even locally after running the changelog generation tasks. So it's not as crucial as I expected it to be when I planned to generate commits to update the changelog automatically. That said, I don't mind adding something to mention this, in case it could motivate people to write more descriptive titles.

* Regarding labels, I don't think contributors can assign labels to PRs. You mean that they mention it in the PR description or something like that?

Reply:
I didn't know contributors could not set a label. I tried on other repos I've contributed on and it doesn't allow me. So the idea for a release is you would have to go through each PR that was merged since the last release and add the appropriate label if needed?

Reply:
Rather based on your other comment, when we review PRs we add the appropriate label then?

Reply:
Yes, only if they include user-visible changes. Otherwise no need to label. I'd like to make this labelling mental process part of the review process. If we do that, we don't need to worry about the changelog anymore.

Reply:
Ok sounds good thanks! I hope we remember. 😆